src/chainstate/stacks/boot/cost-voting.clar

```diff
@@ -0,0 +1,205 @@
+;; The .cost-voting contract
+
+;; error codes
+(define-constant ERR_NO_SUCH_PROPOSAL        1)
+(define-constant ERR_AMOUNT_NOT_POSITIVE     2)
+(define-constant ERR_PROPOSAL_EXPIRED        3)
+(define-constant ERR_PROPOSAL_CONFIRMED      4)
+(define-constant ERR_INSUFFICIENT_FUNDS      5)
+(define-constant ERR_FT_MINT                 6)
+(define-constant ERR_FT_TRANSFER             7)
+(define-constant ERR_STX_TRANSFER            8)
+(define-constant ERR_ALREADY_VETOED          9)
+(define-constant ERR_NOT_LAST_MINER          10)
+(define-constant ERR_INSUFFICIENT_VOTES      11)
+(define-constant ERR_PROPOSAL_VETOED         12)
+(define-constant ERR_FETCHING_BLOCK_INFO     13)
+
+;; cost vote token
+(define-fungible-token cost-vote-token)
+
+;; proposal counters
+(define-data-var proposal-count uint u0)
+(define-data-var confirmed-proposal-count uint u0)
+
+;; cost-function proposals
+(define-map proposals
+    ((proposal-id uint))
+    ((cost-function-contract principal)
```

Comment:
Not a bug, but can you use consistent tuple notation?  I see you use `{ }` below.

Reply:
I think that until #2115 is merged, this syntax needs to be used in `define-map`

Reply:
Done!